generator/internal/rust/templates/crate/src/builders.rs.mustache

```diff
@@ -83,19 +83,28 @@ pub mod {{Codec.ModuleName}} {
             }}
             (*self.0.stub).{{Codec.Name}}(self.0.request, self.0.options).await
         }
-        {{#IsPageable}}
+        {{#Pagination}}
 
         /// Streams the responses back.
         pub async fn paginator(self) -> gax::paginator::Paginator<{{OutputType.Codec.QualifiedName}}, gax::error::Error> {
-            let token = gax::paginator::extract_token(&self.0.request.page_token);
+            use std::clone::Clone;
+            {{#Optional}}
+            let token = match &self.0.request.{{Codec.FieldName}} {
+                std::option::Option::Some(v) => v.clone(),
+                std::option::Option::None => std::string::String::new(),
+            };
```

Comment:
Ditto, this can be `.cloned().unwrap_or_default()`.

Reply:
Done.